src/librustc/sir.rs

```diff
@@ -120,64 +120,66 @@ impl SirFuncCx {
         lvalue: &mir::Place<'_>,
         rvalue: &mir::Rvalue<'_>,
     ) -> ykpack::Statement {
-        match self.lower_place(lvalue) {
-            Ok(lhs) => {
-                let rhs = self.lower_rvalue(rvalue);
-                ykpack::Statement::Assign(ykpack::Place::from(lhs), rhs)
+        let lhs = self.lower_place(lvalue);
+        let rhs = self.lower_rvalue(rvalue);
+        ykpack::Statement::Assign(lhs, rhs)
+    }
+
+    pub fn lower_operand(&self, operand: &mir::Operand<'_>) -> ykpack::Operand {
+        match operand {
+            mir::Operand::Copy(place) | mir::Operand::Move(place) => {
+                ykpack::Operand::Place(self.lower_place(place))
             }
-            Err(_) => ykpack::Statement::Unimplemented(format!("Assign({:?})", lvalue)),
+            mir::Operand::Constant(cst) => ykpack::Operand::Constant(self.lower_constant(cst)),
         }
     }
 
     fn lower_rvalue(&self, rvalue: &mir::Rvalue<'_>) -> ykpack::Rvalue {
         match rvalue {
-            mir::Rvalue::Use(mir::Operand::Copy(place))
-            | mir::Rvalue::Use(mir::Operand::Move(place)) => match self.lower_place(place) {
-                Ok(p) => ykpack::Rvalue::Use(ykpack::Operand::from(p)),
-                Err(_) => ykpack::Rvalue::Unimplemented(format!("Use({:?})", place).to_string()),
-            },
-            mir::Rvalue::Use(mir::Operand::Constant(box constant)) => {
-                match self.lower_constant(constant) {
-                    Ok(c) => ykpack::Rvalue::Use(ykpack::Operand::Constant(c)),
-                    Err(_) => {
-                        ykpack::Rvalue::Unimplemented(format!("Use({:?})", constant).to_string())
-                    }
-                }
-            }
-            _ => ykpack::Rvalue::Unimplemented(format!("{:?}", rvalue).to_string()),
+            mir::Rvalue::Use(opnd) => ykpack::Rvalue::Use(self.lower_operand(opnd)),
+            _ => ykpack::Rvalue::Unimplemented(format!("unimplemented rvalue: {:?}", rvalue)),
         }
     }
 
-    fn lower_place(&self, place: &mir::Place<'_>) -> Result<ykpack::Local, ()> {
-        if place.projection.len() == 0 {
-            Ok(self.lower_local(place.local))
-        } else {
-            // For now we are not dealing with projections.
-            Err(())
+    pub fn lower_place(&self, place: &mir::Place<'_>) -> ykpack::Place {
+        ykpack::Place {
+            local: self.lower_local(place.local),
+            // FIXME projections not yet implemented.
+            projection: place
+                .projection
+                .iter()
+                .map(|p| ykpack::PlaceElem::Unimplemented(format!("{:?}", p)))
+                .collect(),
         }
     }
 
-    fn lower_local(&self, local: mir::Local) -> ykpack::Local {
+    pub fn lower_local(&self, local: mir::Local) -> ykpack::Local {
```

Comment:
Any reason why some of these lowerings are public and others aren't? I feel like it should be the same for all.

Reply:
In some cases I needed access from `codegen_ssa`, but I'll double check them all.

Reply:
Ah I see. I wonder if we should just make them all public then. It seems weird for some to be public and others not.

Reply:
`lower_local()` can be private, but the others need to remain public. Change pushed.

Reply:
So you don't think we should just make them all public? It seems weird for some to be public and others not.

Reply:
@vext01 ping

Reply:
Having things private where possible is a good thing I reckon. It stops people calling things they don't need to or shouldn't.

Reply:
Since this is the same codebase anyways I guess it doesn't matter too much for now.